File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    int coin = 0;
    
    do
    {
        change = round(get_float("Enter change: ") * 100); //round the input to the nearest penny
    }
    while (change < 0.100); //Cheking if input is negative number
    
    if (change >= 25) //Always check if change is greater than 25
    {
        do
        {
            coin++; //add 1 count to the coin variable
            change -= 25;
        }
        while (change >= 25);
    }
    
    if (change >= 10) //Always check if change is greater than 10
    {
        do
        {
            coin++;
            change -= 10;
        }
        while (change >= 10);
    }
    
    if (change >= 5) //Always check if change is greater than 5
    {
        do
        {
            coin++;
            change -= 5;
        }
        while (change >= 5);
    }
    
    if (change >= 1) //Always check if change is greater than 1
    {
        do
        {
            coin++;
            change -= 1 ;
        }
        while (change >= 1);
    }
    printf("%i \n", coin);
}

File: pset6/cash/cash.py
import sys
from cs50 import get_float


def main():

    float_change = get_float("Enter change: ")
    coin_count = 0

    # Ensure correct input
    while float_change <= 0:
        float_change = get_float("Enter change: ")

    int_change = int(float_change * 100)  # convert the change into int datatype

    while(int_change >= 25):  # Always check if change is greater than 25
        int_change -= 25
        coin_count += 1   # add 1 count to the coin variable

    while(int_change >= 10 and int_change < 25):  # Always check if change is between 10 and 25
        int_change -= 10
        coin_count += 1

    while(int_change >= 5 and int_change < 10):  # Always check if change is between 5 and 10
        int_change -= 5
        coin_count += 1

    while(int_change >= 1 and int_change < 5):  # Always check if change is between 1 and 5
        int_change -= 1
        coin_count += 1

    print("Coins:", coin_count)


main()

